File: clients/php/test/Auth/UserAuthContextTest.php
<?php declare(strict_types=1);

namespace Eternaltwin\Test\Auth;

use Eternaltwin\Auth\AuthScope;
use Eternaltwin\Auth\UserAuthContext;
use Eternaltwin\Test\SerializationTestItem;
use Eternaltwin\User\ShortUser;
use Eternaltwin\User\UserDisplayName;
use Eternaltwin\User\UserDisplayNameVersion;
use Eternaltwin\User\UserDisplayNameVersions;
use Eternaltwin\User\UserId;
use PHPUnit\Framework\TestCase;

final class UserAuthContextTest extends TestCase {
  /**
   * @dataProvider provideFromJson
   * @param SerializationTestItem $item
   */
  public function testFromJson(SerializationTestItem $item): void {
    $actual = UserAuthContext::fromJson($item->getJson());
    $this->assertEquals($item->getValue(), $actual);
  }

  public function provideFromJson(): array {
    return SerializationTestItem::fromTestDir(
      "core/auth/user-auth-context",
      [
        "demurgos" => new UserAuthContext(
          AuthScope::Default(),
          new ShortUser(
            UserId::fromString("9f310484-963b-446b-af69-797feec6813f"),
            new UserDisplayNameVersions(new UserDisplayNameVersion(new UserDisplayName("Demurgos")), []),
          ),
          true,
        ),
      ],
    );
  }
}

File: clients/kotlin/src/test/kotlin/net/eternaltwin/auth/UserAuthContextTest.kt
package net.eternaltwin.auth;

import net.eternaltwin.SerializationTestItem
import net.eternaltwin.user.*
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream
import kotlin.test.assertEquals

class UserAuthContextTest {
  companion object {
    @JvmStatic
    fun fromJsonString(): Stream<SerializationTestItem<UserAuthContext>> =
      SerializationTestItem.streamFromTestDir(
        "core/auth/user-auth-context",
        mapOf(
          "demurgos" to UserAuthContext(
            scope = AuthScope.Default,
            user = ShortUser(
              id = UserId("9f310484-963b-446b-af69-797feec6813f"),
              displayName = UserDisplayNameVersions(UserDisplayNameVersion(UserDisplayName("Demurgos"))),
            ),
            isAdministrator = true
          ),
        ),
      )
  }

  @ParameterizedTest
  @MethodSource
  fun fromJsonString(item: SerializationTestItem<UserAuthContext>) {
    val actual = UserAuthContext.fromJsonString(item.jsonString)
    assertEquals(item.value, actual)
  }
}
